### DISCUSSION TITLE
TFT eSPI ST7735 Config issues with Seeed Xiao Rp2040

### DISCUSSION THREAD

**Discussion by git2212** [Bodmer/TFT_eSPI#2617] [General]
Hello,

Having a bit of an issue getting the setup (as indicated in the title) working. Hard to tell what the proper pin selection aught to be for this board with limited number of pins (see [https://wiki.seeedstudio.com/XIAO-RP2040-with-Arduino/](https://wiki.seeedstudio.com/XIAO-RP2040-with-Arduino/) however, hardware SPI would be desirable but not necessary if DMA is still available regardless of pin selection. 

I figure my configuration in the platformio.ini is not quite up to snuff yet but can't determine what I am still missing. Please note I am running the earlephilhower core, no compiler warnings, etc. I know this display works with an ST32 Blue Pill with the rendering code.

The best result I got was a completely white screen (but nothing else) using:

    -D RP2040_PIO_SPI=1
    -D ST7735_DRIVER=1
    -D ST7735_GREENTAB=1
    -D TFT_WIDTH=128
    -D TFT_HEIGHT=160

    -D TFT_MISO=PIN_SPI0_MISO
    -D TFT_MOSI=PIN_SPI0_MOSI
    -D TFT_SCLK=PIN_SPI0_SCK
    -D TFT_CS=PIN_SERIAL1_RX
    -D TFT_DC=PIN_SERIAL1_TX
    -D TFT_RST=A3
    -D TFT_BL=A2

      ;-D TOUCH_CS=22
    -D LOAD_GLCD=1
    -D LOAD_FONT2=1
    -D LOAD_FONT4=1
    -D LOAD_FONT6=1
    -D LOAD_FONT7=1
    -D LOAD_FONT8=1
    -D LOAD_GFXFF=1
    -D SMOOTH_FONT=1

Am I using the wrong pin designations/definitions?... I've tried as an example using the Dxx type pin definitions but that way I get a black screen, even the backlight does not turn on? For example post https://github.com/Bodmer/TFT_eSPI/discussions/1533 seems to indicate that this might be the case.

Any help would be appreciated.

Thx.

**Comment by git2212** [comment]
... just one addition if it makes a difference, I am using this LCD [http://www.lcdwiki.com/1.44inch_SPI_Module_ST7735S_SKU:MSP1443](http://www.lcdwiki.com/1.44inch_SPI_Module_ST7735S_SKU:MSP1443)

**Reply by Bodmer** [reply]
If the correct board has been selected in the IDE then using "PIN_SPI0_MISO" should work fine. If you have selected a generic RP2040 board then these pin references may not be defined. The safe way to define pins is to use the GPIO numbers:
```
-D TFT_MISO=4
-D TFT_MOSI=3
-D TFT_SCLK=2
-D TFT_CS=1
-D TFT_DC=0
-D TFT_RST=29
```

The display link indicates it is 128x128 so correct as follows:
```
-D TFT_HEIGHT=128
```
The display LED pin should be connected to 3.3V (as recommended on the web page). A GPIO pin cannot provide the required current, so the processor pin will be overloaded. A GPIO can only be used if the display incorporates a transistor switch, which that display does not have.

There are different display configuration options, see here:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L102-L111

The SPI clock rate must be specified:
```
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L102-L111
```

The ST7735_GREENTAB128 might be a good one to try, but these displays have multiple vendors and so try the others.  Use a simple sketch such as this one to test:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Test%20and%20diagnostics/Colour_Test

**Comment by git2212** [comment]
Sorry, I linked in the wrong product info, I do have the 1.8" version which is 128x160, and to correct the record for future readers, obtained from [https://www.amazon.ca/gp/product/B07BFV69DZ/ref=ppx_od_dt_b_asin_title_s00?ie=UTF8&psc=1](https://www.amazon.ca/gp/product/B07BFV69DZ/ref=ppx_od_dt_b_asin_title_s00?ie=UTF8&psc=1), mysteriously identified as product id: ‎Walfrontxgez9t1dmf

So the setup with the PR2040 pico seems to be working using the hardcore GPIO pin numbers. The display BL on my board is happily driven by the GPIO pin on my board so didn't have to do a direct connect to 3.3V.

 This thing with the pins is a major pain in the ***, and can make us all look like idiots! One would reasonably assume that picking the right board in the IDE then specifying any version of the pin definitions would work.  My above comment is not an issue with the library per se and is more of a complex problem as PIN numbers and definitions are just a mess and certainly not consistent even across the line of boards from the same maker... how the pins on the board silkscreens are identified, or I should say mislabeled, is also another convo alltogether. Well, rant over, lol, now I can move right along and figure out the rest... I'll start another thread on that as it seems to me it is not related to pins and connectivity.
 
 Thx.
